Python/wolfssl-python-3.8.5.patch

```diff
@@ -0,0 +1,1995 @@
+                                        chatty=False,
+                                        sni_name='supermessage')
+-        self.assertEqual(cm.exception.reason, 'TLSV1_ALERT_ACCESS_DENIED')
++        if ssl.IS_WOLFSSL:
++            self.assertEqual(cm.exception.reason, 'fatal error')
++        else:
++            self.assertEqual(cm.exception.reason, 'TLSV1_ALERT_ACCESS_DENIED')
+ 
+     @needs_sni
+     def test_sni_callback_raising(self):
+@@ -4182,9 +4428,11 @@
+                 stats = server_params_test(client_context, server_context,
+                                            chatty=False,
+                                            sni_name='supermessage')
+-
+-            self.assertEqual(cm.exception.reason,
+-                             'SSLV3_ALERT_HANDSHAKE_FAILURE')
++            if ssl.IS_WOLFSSL:
++                self.assertEqual(cm.exception.reason, 'fatal error')
++            else:
++                self.assertEqual(cm.exception.reason,
++                                'SSLV3_ALERT_HANDSHAKE_FAILURE')
+             self.assertEqual(catch.unraisable.exc_type, ZeroDivisionError)
+ 
+     @needs_sni
+@@ -4203,20 +4451,30 @@
+                                            chatty=False,
+                                            sni_name='supermessage')
+ 
+-
+-            self.assertEqual(cm.exception.reason, 'TLSV1_ALERT_INTERNAL_ERROR')
++            if ssl.IS_WOLFSSL:
++                self.assertEqual(cm.exception.reason, 'fatal error')
++            else:
++                self.assertEqual(cm.exception.reason, 'TLSV1_ALERT_INTERNAL_ERROR')
+             self.assertEqual(catch.unraisable.exc_type, TypeError)
+ 
+     def test_shared_ciphers(self):
+         client_context, server_context, hostname = testing_context()
+-        client_context.set_ciphers("AES128:AES256")
+-        server_context.set_ciphers("AES256")
++        if ssl.IS_WOLFSSL:
++            # wolfSSL doesn't support cipher rules.
++            client_context.set_ciphers("ECDHE-RSA-AES256-GCM-SHA384")
++            server_context.set_ciphers("ECDHE-ECDSA-AES256-GCM-SHA384")
++        else:
++            client_context.set_ciphers("AES128:AES256")
++            server_context.set_ciphers("AES256")
+         expected_algs = [
+             "AES256", "AES-256",
+             # TLS 1.3 ciphers are always enabled
+             "TLS_CHACHA20", "TLS_AES",
+         ]
+ 
++        if ssl.IS_WOLFSSL:
++            expected_algs.append('AES_256')
++
+         stats = server_params_test(client_context, server_context,
+                                    sni_name=hostname)
+         ciphers = stats['server_shared_ciphers'][0]
+@@ -4267,20 +4525,26 @@
+         self.assertTrue(session.id)
+         self.assertGreater(session.time, 0)
+         self.assertGreater(session.timeout, 0)
+-        self.assertTrue(session.has_ticket)
++        if not ssl.IS_WOLFSSL:
++            #wolfSSL is not enabled session ticket as default
```

Comment:
Add space between `#` and comment text here and in other cases below. Reword to "wolfSSL does not enable session tickets by default."